### ISSUE TITLE
Conditional gradient css rules

### ISSUE THREAD

**Issue by joyser** [jbetancur/react-data-table-component#632]
## Feature Check list

- [x] Agree to the [Code of Conduct](https://github.com/jbetancur/react-data-table-component/blob/master/CODE-OF-CONDUCT.md)
- [x] Read the README to ensure the feature is not already present
- [x] You read [Creating Issues, Features and Pull Requests](https://github.com/jbetancur/react-data-table-component/issues/387)
- [x] Considered the value versus complexity for all users of the library as well as library maintenance
- [x] Considered if this can be a storybook or documentation example

## Describe the solution you'd like

I'd like to be able to apply a gradient conditional fill to a cell based on `row.field`

<img width="256" alt="Screenshot 2020-07-22 at 11 48 57" src="https://user-images.githubusercontent.com/3827703/88167859-62a96380-cc11-11ea-9f53-0985d0379a4c.png">

## Describe alternatives you've considered

I don't think this is currently achievable because row isn't accessible within the style object

```javascript
const conditionalRowStyles = [
  {
    when: row => row.profit > 10,
    style: {
      backgroundColor: 'rgba(0, 0, 255, row.profit_margin)',
    },
  }
];

```

- My workaround is to create 4 conditional cell/row rules to bucket the row.profit values, but obviously doesn't get the full gradient effect.

## Additional context

related
https://github.com/jbetancur/react-data-table-component/issues/293#issuecomment-552249458

**Comment by jbetancur** [comment]
Oh, that's an interesting catch. I'll need to investigate if that can be done (I suspect it can)

**Comment by jbetancur** [comment]
@joyser This is now implemented in v6.11.0
